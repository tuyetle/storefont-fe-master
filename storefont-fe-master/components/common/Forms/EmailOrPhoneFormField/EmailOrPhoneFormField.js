import React from 'react';
import { translate } from 'react-i18next';
import PropTypes from 'prop-types';
import { isNil } from 'lodash';
import {
  REQUIRED,
  WRONG_FORMAT,
} from '~/services/validations/validationTypes';
import TextFormField from '~/components/common/Forms/TextFormField/TextFormField';
import validateError from './EmailOrPhoneFormFieldValidation';

import styles from './styles';

const EmailOrPhoneFormField = (props) => {
  const {
    error,
    setError,
  } = props.fieldApi;

  const {
    className,
    errorClass,
    t,
  } = props;
  let errorMsg = '';
  if (error) {
    switch (error) {
    case REQUIRED:
      errorMsg = t('common:emailOrPhoneRequired');
      break;
    case WRONG_FORMAT:
      errorMsg = t('common:wrongFormat');
      break;
    default:
      errorMsg = error;
    }
  }
  const onBlurHandler = (e) => {
    const validation = validateError(e.target.value);
    if (validation) {
      setError(validation);
    }
  };

  const errorClassname = error ? (errorClass || `${styles.error}`) : null;

  return (
    <TextFormField {...props} className={isNil(errorClassname) ? className : `${errorClassname} ${props.className}`} errorMessage={errorMsg} onBlur={onBlurHandler} placeholder={t('auth:emailOrPhone')} />
  );
};
EmailOrPhoneFormField.propTypes = {
  className: PropTypes.string,
  errorClass: PropTypes.string,
  fieldApi: PropTypes.object,
  isRequired: PropTypes.bool,
  t: PropTypes.func,
};
EmailOrPhoneFormField.defaultProps = {
  isRequired: false,
  className: null,
  errorClass: null,
  fieldApi: null,
  t: null,
};

export default translate(['common', 'auth'])(EmailOrPhoneFormField);
