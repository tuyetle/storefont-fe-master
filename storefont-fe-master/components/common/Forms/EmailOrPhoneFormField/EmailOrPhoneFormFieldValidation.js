import { isEmpty } from 'lodash';
import validateEmail from '~/services/validations/validateEmail';
import validatePhone from '~/services/validations/validatePhone';
import {
  REQUIRED,
  WRONG_FORMAT,
} from '~/services/validations/validationTypes';

const validateError = (value) => {
  let error;
  if (!isEmpty(value) && !validateEmail(value) && !validatePhone(value)) {
    error = WRONG_FORMAT;
  }

  return error;
};

export default validateError;
