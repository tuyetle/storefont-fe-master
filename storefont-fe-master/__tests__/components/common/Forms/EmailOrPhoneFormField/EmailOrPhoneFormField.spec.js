
import React from 'react';
import EmailOrPhoneFormField from '~/components/common/Forms/EmailOrPhoneFormField/EmailOrPhoneFormField';
import renderer from 'react-test-renderer';

describe('(Component) EmailOrPhoneFormField', () => {
  let component;
  let props;

  beforeEach(() => {
    props = {
      className: 'test',
      errorClass: 'error',
      t: value => value,
      fieldApi: {
        setError() {},
        setValue() {},
      },
    };
    component = <EmailOrPhoneFormField {...props} />;
  });

  it('should render conrectly', () => {
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
