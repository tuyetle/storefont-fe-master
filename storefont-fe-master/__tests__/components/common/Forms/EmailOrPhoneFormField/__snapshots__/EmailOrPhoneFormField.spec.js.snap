// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) EmailOrPhoneFormField should render conrectly 1`] = `
<input
  className="test null"
  id=""
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  placeholder="auth:emailOrPhone"
  type="text"
  value=""
/>
`;
